dtfs-central-api/api-tests/v1/utilisation-reports/post-validate-utilisation-report-data.api-test.ts

```diff
@@ -73,39 +102,74 @@ describe(`POST ${URL}`, () => {
-        // Arrange
-        const requestBody = getRequestBody();
-
-        const tfmFacility: TfmFacility = {
-          ...aTfmFacility(),
-          facilitySnapshot: {
-            ...aFacility(),
-            ukefFacilityId,
+      describe('and when the supplied facility id exists in the TFM facilities collection', () => {
+        it.each(difference(Object.values(FACILITY_TYPE), Object.values(GEF_FACILITY_TYPE)))(
+          'should return the "facility ID not recognised" error for unsupported facility type "%s"',
+          async (facilityType: FacilityType) => {
+            // Arrange
+            const requestBody = getRequestBody();
+
+            const tfmFacility: TfmFacility = {
+              ...aTfmFacility(),
+              facilitySnapshot: {
+                ...aFacility(),
+                type: facilityType,
+                ukefFacilityId,
+              },
+            };
+
+            const tfmFacilitiesCollection = await mongoDbClient.getCollection(MONGO_DB_COLLECTIONS.TFM_FACILITIES);
+            await tfmFacilitiesCollection.insertOne(tfmFacility);
+
+            // Act
+            const response: SuccessResponse = await testApi.post(requestBody).to(URL);
+
+            // Assert
+            const errorMessages = response.body.csvValidationErrors.map(({ errorMessage }) => errorMessage);
+            expect(errorMessages).toContain(expectedError);
           },
-        };
-
-        const tfmFacilitiesCollection = await mongoDbClient.getCollection(MONGO_DB_COLLECTIONS.TFM_FACILITIES);
-        await tfmFacilitiesCollection.insertOne(tfmFacility);
-
-        // Act
-        const response: SuccessResponse = await testApi.post(requestBody).to(URL);
-
-        // Assert
-        const errorMessages = response.body.csvValidationErrors.map(({ errorMessage }) => errorMessage);
-        expect(errorMessages).not.toContain(expectedError);
+        );
+
+        it.each(Object.values(GEF_FACILITY_TYPE))(
+          'should not return the "facility ID not recognised" error for supported facility type "%s"',
+          async (facilityType: FacilityType) => {
+            // Arrange
+            const requestBody = getRequestBody();
+
+            const tfmFacility: TfmFacility = {
+              ...aTfmFacility(),
+              facilitySnapshot: {
+                ...aFacility(),
+                type: facilityType,
+                ukefFacilityId,
+              },
+            };
+
+            const tfmFacilitiesCollection = await mongoDbClient.getCollection(MONGO_DB_COLLECTIONS.TFM_FACILITIES);
+            await tfmFacilitiesCollection.insertOne(tfmFacility);
+
+            // Act
+            const response: SuccessResponse = await testApi.post(requestBody).to(URL);
+
+            // Assert
+            const errorMessages = response.body.csvValidationErrors.map(({ errorMessage }) => errorMessage);
+            expect(errorMessages).not.toContain(expectedError);
```

Comment:
Do we have a test ensuring that the facilities are returned, rather than "does not contain errors"?

Reply:
This validation endpoint only returns the list of validation errors for cells (as `csvValidationErrors`) so I don't think this is a concern